lock/cond.go

```diff
@@ -0,0 +1,162 @@
+package lock
+
+import (
+	"container/list"
+	"context"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type Ticket interface {
+	// Release releases the ticket. The ticket must not be released multiple times.
```

Comment:
We are (transitively) violating this in the ping controller with `defer ticket.Release()` followed by `ticket.Wait()`.

Reply:
I don't see how it's released multiple times?

Reply:
Because `Waiter()` is called which calls `Release` (as I recall).